File: java/src/test/java/bouchonnois/tests/doubles/PartieDeChasseRepositoryForTests.java
package bouchonnois.tests.doubles;

import bouchonnois.domain.PartieDeChasse;
import bouchonnois.repository.PartieDeChasseRepository;
import lombok.Getter;

import java.util.HashMap;
import java.util.Map;
import java.util.UUID;

public class PartieDeChasseRepositoryForTests implements PartieDeChasseRepository {
    private final Map<UUID, PartieDeChasse> partiesDeChasse = new HashMap<>();
    @Getter
    private PartieDeChasse savedPartieDeChasse;

    @Override
    public void save(PartieDeChasse partieDeChasse) {
        savedPartieDeChasse = partieDeChasse;
        partiesDeChasse.put(partieDeChasse.getId(), partieDeChasse);
    }

    @Override
    public PartieDeChasse getById(UUID partieDeChasseId) {
        return null;
    }
}

File: kotlin/src/test/kotlin/bouchonnois/tests/doubles/PartieDeChasseRepositoryForTests.kt
package bouchonnois.tests.doubles

import bouchonnois.domain.PartieDeChasse
import bouchonnois.repository.PartieDeChasseRepository
import java.util.*

class PartieDeChasseRepositoryForTests : PartieDeChasseRepository {
    var savedPartieDeChasse: PartieDeChasse? = null
    private val partiesDeChasse: MutableMap<UUID, PartieDeChasse> = mutableMapOf()

    override fun save(partieDeChasse: PartieDeChasse) {
        savedPartieDeChasse = partieDeChasse
        add(partieDeChasse)
    }

    override fun getById(partieDeChasseId: UUID?): PartieDeChasse? =
        partiesDeChasse[partieDeChasseId]

    fun add(partieDeChasse: PartieDeChasse) {
        partiesDeChasse.put(partieDeChasse.id!!, partieDeChasse)
    }
}